public/js/reader.js

```diff
@@ -1,408 +1,601 @@
+    $(document).on("click.toggle_header", "#toggle-header input", Reader.toggleHeader);
+    $(document).on("click.toggle_progress", "#toggle-progress input", Reader.toggleProgressTracking);
+    $(document).on("click.toggle_infinite_scroll", "#toggle-infinite-scroll input", Reader.toggleInfiniteScroll);
+    $(document).on("submit.container_width", "#container-width-input", Reader.registerContainerWidth);
+    $(document).on("click.container_width", "#container-width-apply", Reader.registerContainerWidth);
+    $(document).on("click.pagination_change_pages", ".page-link", Reader.handlePaginator);
+
+    $(document).on("click.close_overlay", "#overlay-shade", closeOverlay);
+    $(document).on("click.toggle_archive_overlay", "#toggle-archive-overlay", Reader.toggleArchiveOverlay);
+    $(document).on("click.toggle_settings_overlay", "#toggle-settings-overlay", Reader.toggleSettingsOverlay);
+    $(document).on("click.toggle_help", "#toggle-help", Reader.toggleHelp);
+    $(document).on("click.regenerate_thumbnail", "#regenerate-thumbnail", Reader.regenerateThumbnail);
+    $(document).on("click.regenerate_archive_cache", "#regenerate-cache", () => {
+        window.location.href = `./reader?id=${Reader.id}&force_reload=1`;
+    });
+    $(document).on("click.edit_metadata", "#edit-archive", () => openInNewTab(`./edit?id=${Reader.id}`));
+    $(document).on("click.add_category", "#add-category", () => addArchiveToCategory(Reader.id, $("#category").val()));
+
+    // check and display warnings for unsupported filetypes
+    Reader.checkFiletypeSupport();
+
+    // remove the "new" tag with an api call
+    clearNew(Reader.id);
+
+    Reader.infiniteScroll = localStorage.infiniteScroll === "true" || false;
+    $(Reader.infiniteScroll ? "#infinite-scroll-on" : "#infinite-scroll-off").addClass("toggled");
+
+    $(document).on("click.thumbnail", ".quick-thumbnail", (e) => {
+        closeOverlay();
+        const pageNumber = +$(e.target).closest("div[page]").attr("page");
+        if (Reader.infiniteScroll) {
+            $("#display img").get(pageNumber).scrollIntoView({ behavior: "smooth" });
+        } else {
+            Reader.goToPage(pageNumber);
+        }
+    });
+
+    if (Reader.infiniteScroll) {
+        Reader.initInfiniteScrollView();
+        return;
+    }
+
+    // all these init values need to be after the infinite loader check
+    $(document).on("click.imagemap_change_pages", "#Map area", Reader.handlePaginator);
+    $(window).on("resize", Reader.updateImagemap);
+
+    let params = new URLSearchParams(window.location.search);
+
+    // initialize current page and bind popstate
+    $(window).on("popstate", () => {
+        params = new URLSearchParams(window.location.search);
+        if (params.has("p")) {
+            const paramsPage = +params.get("p");
+            Reader.goToPage(paramsPage - 1, true);
+        }
+    });
+
+    Reader.currentPage = +params.get("p") - 1;
+    Reader.currentPage ||= (
```

Comment:
This seems to be broken: If I have `Reader.progress` set to a server-side number and `Reader.ignoreProgress` set to false, the reader still boots me to the first page when opening an url without a `p` query parameter. Not sure why.

Reply:
Fixed, added a comment too. The issue was that if p is null then it's coerced to 0, and currentPage becomes - 1, so it's not catched by the OR assignment.